packages/ember-cli-eyeglass/src/index.ts

```diff
@@ -77,10 +83,72 @@ function localEyeglassAddons(addon): Array<{path: string}> {
   return paths;
 }
 
+interface EyeglassProjectInfo {
+  apps: Array<any>;
+}
+interface EyeglassAddonInfo {
+  name: string;
+  parentPath: string;
+  isApp: boolean;
+  app: any; // is this safe to cache across builds?
+}
+interface EyeglassAppInfo {
+  assets: BroccoliSymbolicLinker;
+  sessionCache: Map<string, string | number>;
+}
+
 const EMBER_CLI_EYEGLASS = {
   name: 'ember-cli-eyeglass',
+  included(parent) {
+    this._super.included.apply(this, arguments);
+    this.initSelf();
+  },
+  initSelf() {
+    if (EYEGLASS_INFO_PER_ADDON.has(this)) return;
+    let app = findHost(this);
+    if (!app) return;
+    let isApp = (this.app === app);
+    let name = app.name;
+    if (!isApp) {
+      let thisName = typeof this.parent.name === "function" ? this.parent.name() : this.parent.name;
+      name = `${name}/${thisName}`
+    }
+    let parentPath = this.parent.root;
+    if (isApp) {
+      apps.push(app);
+      // we create the symlinker in persistent mode because there's not a good
+      // way yet to recreate the symlinks when sass files are cached. I would
+      // worry about it more but it seems like the dist directory is cumulative
+      // across builds anyway.
+      EYEGLASS_INFO_PER_APP.set(app, {
+        sessionCache: new Map(),
+        assets: new BroccoliSymbolicLinker({}, {annotation: app.name, persistentOutput: true})
+      });
+    }
+    let addonInfo = {isApp, name, parentPath, app};
+    EYEGLASS_INFO_PER_ADDON.set(this, addonInfo);
+  },
   postBuild(result) {
     Eyeglass.resetGlobalCaches();
+    for (let app of apps) {
+      let appInfo = EYEGLASS_INFO_PER_APP.get(app);
+      if (appInfo) {
+        appInfo.assets.reset();
+        appInfo.sessionCache.clear();
+      } else {
+        // eslint-disable-next-line no-console
+        console.warn("eyeglass app info is missing during postBuild.");
```

Comment:
Is this actionable for the user? If so, we should likely use `this.ui` which should be an instance of https://github.com/ember-cli/console-ui and facilitates a way for unified output.

If not, then we should likely not output it by default, rather then debug logging.

Reply:
This doesn't happen anymore. I'll remove it. But I'm glad to learn about `this.ui`, there's probably other places in my code that should be using it.